src/Symfony/Component/Form/Extension/DataCollector/FormDataCollector.php

```diff
@@ -221,6 +238,85 @@ public function getData()
+                continue;
+            }
+            $form['view_vars'] = array_map($cloneVar, $form['view_vars']);
+            $form['type_class'] = $cloneVar(new ClassStub($form['type_class']));
+            $form['synchronized'] = $cloneVar($form['synchronized']);
+            $form['passed_options'] = array_map($cloneVar, $form['passed_options']);
+            $form['resolved_options'] = array_map($cloneVar, $form['resolved_options']);
+            $form['default_data'] = array_map($cloneVar, $form['default_data']);
+            $form['submitted_data'] = array_map($cloneVar, $form['submitted_data']);
+
+            if (!empty($form['errors'])) {
+                foreach ($form['errors'] as $i => $error) {
+                    if (!empty($error['trace'])) {
+                        $form['errors'][$i]['trace'] = array_map($cloneVar, $error['trace']);
+                    }
+                }
+            }
+        }
+
+        return serialize($this->data);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function cloneVar($var)
+    {
+        if (null === $this->cloner) {
+            if (class_exists(ClassStub::class)) {
+                $this->cloner = new VarCloner();
+                $this->cloner->setMaxItems(25);
+                $this->cloner->addCasters(array(
+                    Stub::class => function (Stub $v, array $a, Stub $s, $isNested) {
+                        return $isNested ? $a : StubCaster::castStub($v, $a, $s, true);
+                    },
+                    \Exception::class => function (\Exception $e, array $a, Stub $s) {
+                        if (isset($a[$k = "\0Exception\0previous"])) {
+                            unset($a[$k]);
+                            ++$s->cut;
+                        }
+
+                        return $a;
+                    },
+                    FormInterface::class => function (FormInterface $f, array $a) {
+                        return array(
+                            Caster::PREFIX_VIRTUAL.'name' => $f->getName(),
+                            Caster::PREFIX_VIRTUAL.'type_class' => new ClassStub(get_class($f->getConfig()->getType()->getInnerType())),
+                        );
+                    },
+                    ConstraintViolationInterface::class => function (ConstraintViolationInterface $v, array $a) {
+                        return array(
+                            Caster::PREFIX_VIRTUAL.'root' => $v->getRoot(),
+                            Caster::PREFIX_VIRTUAL.'path' => $v->getPropertyPath(),
+                            Caster::PREFIX_VIRTUAL.'value' => $v->getInvalidValue(),
+                        );
+                    },
+                ));
+            } else {
+                @trigger_error(sprintf('Using the %s() method without the VarDumper component is deprecated since version 3.2 and won\'t be supported in 4.0. Install symfony/var-dumper version 3.2 or above.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
this will say that using ``FormDataCollector::cloneVar`` without the component is deprecated, but this is a private method. You should probably talk about the public API triggering it instead